firestore/cloud-async-client/snippets.py

```diff
@@ -20,8 +20,9 @@ async def quickstart_new_instance():
     # [START firestore_setup_client_create_async]
     from google.cloud import firestore
 
-    # Project ID is determined by the GCLOUD_PROJECT environment variable
-    db = firestore.AsyncClient()
+    # The `project` parameter is optional and if not supplied will
+    # fall back to your gcloud project configuration.
```

Comment:
I may be too concerned, but the way we infer projects is broader than gcloud. https://github.com/googleapis/python-cloud-core/blob/master/google/cloud/client.py#L237

```
    # If you don't specify credentials when constructing the client, the
    # client library will look for credentials in the environment.
```

Is how we describe it in auth, https://cloud.google.com/docs/authentication/getting-started#verifying_authentication. Even that isn't very specific to what is happening though, maybe intentionally.

Reply:
I share this concern - went back and forth on it with @kolea2 last week on the phrasing. Maybe the three of us can put our heads together on one succinct phrasing to rule them all.

Reply:
I think you still need to update this comment to match the other?

Reply:
Oh my goodness - this Python a/sync split will be the end of me one day.